packages/typespec-test/test/batch_modular/generated/typespec-ts/src/BatchService.ts

```diff
@@ -0,0 +1,815 @@
+    },
+    rebootComputeNode: (
+      poolId: string,
+      nodeId: string,
+      options: ComputeNodesRebootComputeNodeOptions = { requestOptions: {} }
+    ): Promise<void> => {
+      return rebootComputeNode(this._client, poolId, nodeId, options);
+    },
+    reimageComputeNode: (
+      poolId: string,
+      nodeId: string,
+      options: ComputeNodesReimageComputeNodeOptions = { requestOptions: {} }
+    ): Promise<void> => {
+      return reimageComputeNode(this._client, poolId, nodeId, options);
+    },
+    disableScheduling: (
+      poolId: string,
+      nodeId: string,
+      options: ComputeNodesDisableSchedulingOptions = { requestOptions: {} }
+    ): Promise<void> => {
+      return disableScheduling(this._client, poolId, nodeId, options);
+    },
+    enableScheduling: (
+      poolId: string,
+      nodeId: string,
+      options: ComputeNodesEnableSchedulingOptions = { requestOptions: {} }
+    ): Promise<void> => {
+      return enableScheduling(this._client, poolId, nodeId, options);
+    },
+    getRemoteLoginSettings: (
+      poolId: string,
+      nodeId: string,
+      options: ComputeNodesGetRemoteLoginSettingsOptions = {
+        requestOptions: {},
+      }
+    ): Promise<ComputeNodeGetRemoteLoginSettingsResult> => {
+      return getRemoteLoginSettings(this._client, poolId, nodeId, options);
+    },
+    getRemoteDesktop: (
+      poolId: string,
+      nodeId: string,
+      options: ComputeNodesGetRemoteDesktopOptions = { requestOptions: {} }
+    ): Promise<void> => {
+      return getRemoteDesktop(this._client, poolId, nodeId, options);
+    },
+    uploadBatchServiceLogs: (
+      containerUrl: string,
+      startTime: Date,
+      poolId: string,
+      nodeId: string,
+      options: ComputeNodesUploadBatchServiceLogsOptions = {
+        requestOptions: {},
+      }
+    ): Promise<UploadBatchServiceLogsResult> => {
+      return uploadBatchServiceLogs(
+        this._client,
+        containerUrl,
+        startTime,
+        poolId,
+        nodeId,
+        options
+      );
+    },
+    list: (
+      poolId: string,
+      options: ComputeNodesListOptions = { requestOptions: {} }
+    ): Promise<ComputeNodeListResult> => {
+      return list(this._client, poolId, options);
+    },
+  };
+  computeNodeExtensions = {
+    getComputeNodeExtensions: (
+      poolId: string,
+      nodeId: string,
+      extensionName: string,
+      options: ComputeNodeExtensionsGetComputeNodeExtensionsOptions = {
+        requestOptions: {},
+      }
+    ): Promise<NodeVMExtension> => {
+      return getComputeNodeExtensions(
+        this._client,
+        poolId,
+        nodeId,
+        extensionName,
+        options
+      );
+    },
+    listComputeNodeExtensions: (
+      poolId: string,
+      nodeId: string,
+      options: ComputeNodeExtensionsListComputeNodeExtensionsOptions = {
+        requestOptions: {},
+      }
+    ): Promise<NodeVMExtensionList> => {
```

Comment:
I may not get the point here, why the classic method would return the REST model?

According to my understanding the model `NodeVMExtensionList` would only accept the json property names not any renaming.